### ISSUE TITLE
[a11y]: some elements not visible in Windows High contrast mode

### ISSUE THREAD

**Issue by Alienah** [carbon-design-system/carbon#20290]
### Package

@carbon/react

### Operating System

Windows

### Package version

v1.88.0

### React version

v19.1.1

### Automated testing tool and ruleset

IBM Equal Access Accessibility Checker

### Description

Some carbon elements are not visible with Windows high contrast mode. I specify below the affected elements and the changes we made to fix it in our project, but it should be fixed in the components themselves

```
@media (forced-colors: active), screen and (-ms-high-contrast:active) {
  .cds--header__action:focus,
  .cds--header__name:focus,
  button.cds--accordion__heading:focus {
    color: Highlight;
    outline: 1px solid Highlight;
  }

  .cds--btn--ghost:not([disabled]) svg,
  .cds--copy-btn svg,
  .cds--header__menu-trigger svg,
  .cds--progress-step--incomplete svg,
  .cds--toggletip-button svg {
    fill: ButtonText;
  }

  .cds--toggletip-button:hover svg {
    fill: Highlight;
  }

  .cds--toggle__switch--checked {
    background-color: Highlight;
  }

}
```

### Reproduction/example

See the mentioned components in https://react.carbondesignsystem.com/?path=/docs/getting-started-welcome--welcome

### Steps to reproduce

* Use  windows OS
* Change the High contrast mode in the system configuration
* Swi a dark mode 
* Navigate through the carbon components mentioned in this issue and switch between the different carbon themes 
* See the icons, etc are not visible

### Suggested Severity

None

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [ ] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by tay1orjones** [comment]
Thanks for reporting this, it looks like some of this is sort-of a follow up to #19352. Techniques and testing described there might apply here too.

**Comment by MichaelEissner** [comment]
I have the same problem with a disbaled button. I haven't checked yet whether other components are affected.
The “Night Sky” color scheme is active in the image. The button text is not visible in Firefox, while in Chrome and Edge the text is visible but not optimally legible.

<img width="720" height="424" alt="Image" src="https://github.com/user-attachments/assets/8196f5c0-5314-458c-840e-828639f7555f" />

Example is available on the cds website: https://react.carbondesignsystem.com/?path=/story/components-button--default&args=disabled:!true
